spec/support/shared_examples/can_present_carrier_name.rb

```diff
@@ -29,39 +31,30 @@
     end
   end
 
-  let(:registered_name) { Faker::Company.name }
-  let(:trading_name) { Faker::Lorem.sentence(word_count: 3) }
-
   context "with a registered name and without a trading name" do
-    before do
-      registration.registered_company_name = registered_name
-      registration.company_name = nil
-    end
+    let(:registered_name) { Faker::Company.name }
+    let(:company_name) { nil }
 
     it "returns the registered name" do
       expect(subject.carrier_name).to eq registered_name
     end
   end
 
   context "without a registered name and with a trading name" do
-    before do
-      registration.registered_company_name = nil
-      registration.company_name = trading_name
-    end
+    let(:registered_name) { nil }
+    let(:company_name) { Faker::Lorem.sentence(word_count: 3) }
 
     it "returns the trading name" do
-      expect(subject.carrier_name).to eq trading_name
+      expect(subject.carrier_name).to eq company_name
     end
   end
 
   context "with both a registered name and a trading name" do
-    before do
-      registration.registered_company_name = registered_name
-      registration.company_name = trading_name
-    end
+    let(:registered_name) { Faker::Company.name }
+    let(:company_name) { Faker::Lorem.sentence(word_count: 3) }
 
     it "returns the registered name" do
```

Comment:
Could this be
`it "returns the **carrier_name**"`
Assuming 'carrier_name' is the presentation name